tests/e2e/tests/bookinfo/demo_test.go

```diff
@@ -37,22 +37,22 @@ import (
 const (
 	u1                                 = "normal-user"
 	u2                                 = "test-user"
-	bookinfoYaml                       = "samples/bookinfo/kube/bookinfo.yaml"
-	bookinfoRatingsv2Yaml              = "samples/bookinfo/kube/bookinfo-ratings-v2.yaml"
-	bookinfoRatingsMysqlYaml           = "samples/bookinfo/kube/bookinfo-ratings-v2-mysql.yaml"
-	bookinfoDbYaml                     = "samples/bookinfo/kube/bookinfo-db.yaml"
-	bookinfoMysqlYaml                  = "samples/bookinfo/kube/bookinfo-mysql.yaml"
-	bookinfoDetailsExternalServiceYaml = "samples/bookinfo/kube/bookinfo-details-v2.yaml"
+	bookinfoSampleDir                  = "samples/bookinfo"
```

Comment:
if this is still a constant, which it is, I'm a bit unsure what the change does
(I can agree it'd would have been cleaner to have the common prefix separated vs copypasted, but the end result is still the same... so ... what does this part of the change do ?)

Reply:
This is a minor refactoring. The change prevents typing errors in the future files to be added:
```
bookinfoNewYaml = "samles/bookinfo/kube/bookinfo.yaml"
```

vs.

```
bookinfoNewYaml = bookinfoSamleDir + "/kube/bookinfo.yaml"
```

Both code snippets have a typo, _p_ is missing in _sample_.

In the first case, it will not be caught by the compiler and will fail in the test.
In the second case, it will be caught by the compiler, which is better.

Reply:
except if you had a typo in
```
bookinfoSampleDir                  = "samles/bookinfo"
```
it won't either... but hopefully all the tests would

Reply:
yes, but `bookinfoSampleDir` you define only once. In the first case, you risk to make a typo in any new yaml you add.

Reply:
if you use the constant in the code to join the dir with the files, it's also only once; either 100% correct or 100% wrong

Reply:
yep, this is what I did after your proposal - https://github.com/istio/istio/pull/4185/commits/4615a997a879cd1cdc21acbb70cb96eba5465242